jenkins-jobs/docker/debezium-testing-system/Dockerfile

```diff
@@ -1,18 +1,29 @@
-FROM registry.access.redhat.com/ubi8/openjdk-17
-USER root
-RUN microdnf install git   
-RUN microdnf install unzip
+FROM fedora:35
 
-ARG OCP_CLIENT=3.7.23
+RUN dnf -y install git unzip zip findutils
 
+ARG OCP_CLIENT=3.7.23
 RUN curl --retry 7 -Lo /tmp/client-tools.tar.gz "https://mirror.openshift.com/pub/openshift-v3/clients/${OCP_CLIENT}/linux/oc.tar.gz"
-
 RUN tar zxf /tmp/client-tools.tar.gz -C /usr/local/bin oc \
     && rm /tmp/client-tools.tar.gz
 
-RUN microdnf clean all 
+ARG sdkman_init="/root/.sdkman/bin/sdkman-init.sh"
+
+RUN curl -s "https://get.sdkman.io" | bash
+RUN source $sdkman_init &&\
+    sdk install java 17.0.2-open&&\
+    sdk install maven 3.8.4
+
+# this runs when only when dbz repo/branch is changed
+ARG repository
+ARG branch
+
+RUN git clone --branch $branch $repository /testsuite/debezium
+RUN source $sdkman_init && mvn clean install -DskipTests -DskipITs -f /testsuite/debezium/pom.xml
+
 COPY testsuite-deployment.sh /testsuite/testsuite-deployment.sh
-COPY library.sh /testsuite/library.sh
 RUN chmod a+x /testsuite/testsuite-deployment.sh
 
-ENTRYPOINT /testsuite/testsuite-deployment.sh 
+COPY library.sh /testsuite/library.sh
+
+ENTRYPOINT /testsuite/testsuite-deployment.sh
```

Comment:
I would recommend sing multi-staged docker build so that the final image is thinner without all the applications not needed in the final image. Also, the sdkman is not necessarily needed, when installing just one version of java and maven.